### PR TITLE
convex decomposition work with Surface_mesh_item

### LINKED ISSUE #1932: Polyhedron_demo: Use Scene_surface_mesh_item instead of Scene_polyhedron_item
## Issue Details

We should use the Scene_surface_mesh_item where it is possible, as it is more efficient in terms of memory consumption. We should integrate the plugins by package, to dispatch tests and to avoid a too important difference with the current master branch.

Packages:

- [x] AABB_tree
- [x] Convex_decomposition
- [x] Convex_hull
- [x] IO
- [x] Mesh_2
- [ ] Mesh_3
- [x] Operations_on_polyhedra
- [x] PCA
- [ ] PMP
- [ ] Point_set
- Subdivision_methods
- [ ] Surface_mesh
- [x] Surface_mesh_deformation

Plugins: 

- [x] Plugins/AABB_tree/Cut_plugin.cpp
- [x]  Plugins/PCA/Create_bbox_mesh_plugin.cpp
- [x]  Plugins/PCA/Pca_plugin.cpp
- [x]  Plugins/PCA/Scene_polyhedron_transform_item.h
- [x]  Plugins/PCA/Edit_box_plugin.cpp
- [x]  Plugins/Mesh_3/Detect_sharp_edges_plugin.cpp
- [ ]  Plugins/Mesh_3/Optimization_plugin_cgal_code.cpp
- [x]  Plugins/Mesh_3/Mesh_3_plugin.cpp
- [x]  Plugins/IO/GOCAD_io_plugin.cpp
- [x]  Plugins/IO/Surf_io_plugin.cpp
- [x]  Plugins/IO/STL_io_plugin.cpp
- [x]  Plugins/IO/VTK_io_plugin.cpp
- [x]  Plugins/Convex_decomposition/Nef_plugin.cpp
- [x]  Plugins/Surface_mesh_deformation/Edit_polyhedron_plugin.cpp
- [x]  Plugins/Surface_mesh_deformation/Scene_edit_polyhedron_item.h
- [ ]  Plugins/Point_set/Surface_reconstruction_plugin.cpp #Uses a Polyhedron_incremental_builder_3
- [x]  Plugins/Point_set/Point_set_shape_detection_plugin.cpp
- Plugins/Surface_mesh/Remeshing_plugin.cpp # Must enhance Mesh_3 and then remove this plugin
- [x]  Plugins/Surface_mesh/Mesh_simplification_plugin.cpp
- [ ]  Plugins/Surface_mesh/Mesh_segmentation_plugin.cpp 
- [x]  Plugins/Surface_mesh/Shortest_path_plugin.cpp
-  Plugins/Surface_mesh/Remeshing_plugin_cgal_code.cpp # Must enhance Mesh_3 and then remove this plugin
-  Plugins/Surface_mesh/Parameterization_plugin.cpp#There is an on going SF concerning this plugin
- [x]  Plugins/Convex_hull/Convex_hull_plugin.cpp
- [x]  Plugins/Convex_hull/Kernel_plugin.cpp
- [x]  Plugins/Operations_on_polyhedra/Clip_polyhedron_plugin.cpp
- [x]  Plugins/Operations_on_polyhedra/Point_set_from_vertices_plugin.cpp
- [x]  Plugins/PMP/Inside_out_plugin.cpp
- [x]  Plugins/PMP/Point_inside_polyhedron_plugin.cpp
- [x]  Plugins/PMP/Polyhedron_slicer_plugin.cpp
- [x]  Plugins/PMP/Corefinement_plugin.cpp
- [x]  Plugins/PMP/Polyhedron_stitching_plugin.cpp
- [x]  Plugins/PMP/Repair_polyhedron_plugin.cpp
- [x]  Plugins/PMP/Surface_intersection_plugin.cpp
- [x]  Plugins/PMP/Distance_plugin.cpp
- [x]  Plugins/PMP/Selection_plugin.cpp
- [x]  Plugins/PMP/Join_and_split_polyhedra_plugin.cpp
- [x]  Plugins/PMP/Hole_filling_polyline_plugin.cpp
- [x]  Plugins/PMP/Hole_filling_plugin.cpp # TODO check normalize_borders
- [x]  Plugins/PMP/Jet_fitting_plugin.cpp
- [x]  Plugins/PMP/Isotropic_remeshing_plugin.cpp
- [ ]  Plugins/PMP/Mean_curvature_flow_skeleton_plugin.cpp
- [x]  Plugins/PMP/Triangulate_facets_plugin.cpp
- [x]  Plugins/Mesh_2/Mesh_2_plugin.cpp
-  Plugins/Subdivision_methods/Subdivision_methods_plugin.cpp #There is an on going SF concerning this plugin

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes

The Nef_plugin works with surface_mesh_items

This PR is based on #1954 
## Release Management

* Affected package(s):Polyhedron_demo
* Issue(s) solved (if any): fix #1932

### CODE REVIEW

**Comment by lrineau** [comment]
This PR seems simple enough to be reviewed, as soon as it is rebased on the latest version of #1954.

**Comment by maxGimeno** [comment]
This branch is not compatible with #1983 in integration, it should be removed from the tests until #1983 is integrated, and then rebased on master and modified.

**Comment by maxGimeno** [comment]
Ready for testing.

**Comment by lrineau** [comment]
The polyhedron demo is fully red because of this PR:
```
/home/cgal_tester/build/src/cmake/platforms/Fedora-Release/test/Polyhedron_Demo/Scene_nef_polyhedron_item.cpp: In static member function 'static Scene_nef_polyhedron_item* Scene_nef_polyhedron_item::from_polygon_mesh(Scene_surface_mesh_item*)':
/home/cgal_tester/build/src/cmake/platforms/Fedora-Release/test/Polyhedron_Demo/Scene_nef_polyhedron_item.cpp:717:3: error: 'SMesh' is not a member of 'Scene_surface_mesh_item'
   Scene_surface_mesh_item::SMesh* sm = item->polyhedron();
   ^~~~~~~~~~~~~~~~~~~~~~~
/home/cgal_tester/build/src/cmake/platforms/Fedora-Release/test/Polyhedron_Demo/Scene_nef_polyhedron_item.cpp:717:35: error: 'sm' was not declared in this scope
   Scene_surface_mesh_item::SMesh* sm = item->polyhedron();
                                   ^~
```
(quoted from [the `Fedora-Release` column](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.11-Ic-43/Polyhedron_Demo/TestReport_lrineau_Fedora-Release.gz), but see also [the testsuite results](https://cgal.geometryfactory.com/CGAL/testsuite/results-4.11-Ic-43.shtml#Polyhedron_Demo).)

File: Polyhedron/demo/Polyhedron/Plugins/Convex_decomposition/Nef_plugin.cpp [hunk 1]
```diff
@@ -106,26 +111,36 @@ Polyhedron_demo_nef_plugin::on_actionToNef_triggered()
 
   Scene_polyhedron_item* item =
     qobject_cast<Scene_polyhedron_item*>(scene->item(index));
+  Scene_surface_mesh_item* sm_item =
+    qobject_cast<Scene_surface_mesh_item*>(scene->item(index));
+  if(!item && !sm_item)
+    return;
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Plugins/Convex_decomposition/Nef_plugin.cpp]: There is an indentation issue of the whole block after this `return`, and that gives a warning:
```
[ 36%] Building CXX object demo/Polyhedron/Plugins/Convex_decomposition/CMakeFiles/nef_plugin.dir/Nef_plugin.cpp.o
/home/lrineau/Git/cgal/Polyhedron/demo/Polyhedron/Plugins/Convex_decomposition/Nef_plugin.cpp: In member function ‘void Polyhedron_demo_nef_plugin::on_actionToNef_triggered()’:
/home/lrineau/Git/cgal/Polyhedron/demo/Polyhedron/Plugins/Convex_decomposition/Nef_plugin.cpp:116:3: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
   if(!item && !sm_item)
   ^~
/home/lrineau/Git/cgal/Polyhedron/demo/Polyhedron/Plugins/Convex_decomposition/Nef_plugin.cpp:119:5: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘if’
```

**Pull request merged**